cases_new/arithmetic/max.yaml

```diff
@@ -0,0 +1,70 @@
+base_uri: https://github.com/substrait-io/substrait/blob/main/extensions/substrait/extensions/functions_arithmetic.yaml
+function: max
+cases:
+- group: basic
+  description: Basic examples without any special cases
+  tests:
+  - (int_col): 20|i8
```

Comment:
Once we get something nice for scalars, we can come back to aggregates. So deferring comments on this part.